### PR TITLE
Additional tests pagination

### PR DESCRIPTION
**Pull request by kayladnls**
This PR: 

Fix bugs: 
- [x] Request Pipeline/Additional Middleware
- [x] Pagination middlewares

Adds tests to cover the above fixes.

### CODE REVIEW

File: src/Middleware/Request/Pagination/OffsetLimitPagination.php [hunk 1]
```diff
@@ -52,6 +44,36 @@ public function process($payload)
         return $payload;
     }
 
+//    /**
+//     * @param Payload $payload
```
Comment (rcatlin) [inline src/Middleware/Request/Pagination/OffsetLimitPagination.php]: Remove?

File: src/Middleware/Request/RequestedFields.php [hunk 2]
```diff
@@ -22,7 +22,7 @@ class RequestedFields implements StageInterface
     public function process($payload)
```
Comment (rcatlin) [inline src/Middleware/Request/RequestedFields.php]: Can we type-hint?

Comment (kayladnls) [inline src/Middleware/Request/RequestedFields.php]: No. It has to be consistent with the StageInterface.

File: src/Piston.php [hunk 3]
```diff
@@ -101,10 +92,31 @@ public function register(ServiceProvider\AbstractServiceProvider $serviceProvide
...
+    private function processPipeline()
+    {
+        (new PipelineProcessor())->handlePayload($this->getSubject());
+    }
+
+    private function loadContainer()
+    {
+        (new RequestPipeline())->process($this->getSubject());
+
+        $this->container->add('Request', $this->request, true);
```
Comment (rcatlin) [inline src/Piston.php]: Can we use FQCNs as the aliases for the Request and Response?

File: src/Middleware/Request/RequestedFields.php (hunk 2 above)
Comment (rcatlin) [inline src/Middleware/Request/RequestedFields.php]: Got it. I think you've fielded this question before. :stuck_out_tongue_closed_eyes: :

**Comment by rcatlin** [comment]
:+1: pending Travis!

**Pull request merged**